bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieRequestProcessor.java

```diff
@@ -619,7 +619,9 @@ private void processAddRequest(final BookieProtocol.ParsedAddRequest r, final Ch
     }
 
     private void processReadRequest(final BookieProtocol.ReadRequest r, final Channel c) {
-        ReadEntryProcessor read = ReadEntryProcessor.create(r, c, this);
+        ExecutorService fenceThreadPool =
+                null == highPriorityThreadPool ? null : highPriorityThreadPool.chooseThread(c);
+        ReadEntryProcessor read = ReadEntryProcessor.create(r, c, this, fenceThreadPool);
```

Comment:
can we add a test for fenceThreadPool = null case ?

Reply:
I pushed 5cdebf873666da60162853d9380b2e2e99099195